cheroot/server.py

```diff
@@ -1798,18 +1860,28 @@ def bind(self, family, type, proto=0):
         )
         if listening_ipv6:
             try:
-                self.socket.setsockopt(
+                sock.setsockopt(
                     socket.IPPROTO_IPV6, socket.IPV6_V6ONLY, 0)
             except (AttributeError, socket.error):
                 # Apparently, the socket option is not available in
                 # this machine's TCP stack
                 pass
 
-        self.socket.bind(self.bind_addr)
-        # TODO: keep requested bind_addr separate real bound_addr (port is
-        # different in case of ephemeral port 0)
-        self.bind_addr = self.socket.getsockname()
-        if family in (
+        return sock
+
+    @staticmethod
+    def bind_socket(socket_, bind_addr):
+        """Bind the socket to given interface."""
+        socket_.bind(bind_addr)
+        return socket_
+
+    @staticmethod
+    def resolve_real_bind_addr(socket_):
```

Comment:
Why not use `sock`, which is used elsewhere in this module for the `socket.socket` instance?

Reply:
Because it's a function signature, more or less public interface. So I restricted usage of shortcuts inside functions only,